modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/title_editor/LocalizedDropdown.es.js

```diff
@@ -0,0 +1,184 @@
+import ClayIcon from '../shared/ClayIcon.es';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class LocalizedDropdown extends React.Component {
+	static defaultProps = {
+		availableLanguages: []
+	}
+
+	static propTypes = {
+		availableLanguages: PropTypes.array,
+		initialLang: PropTypes.string,
+		initialOpen: PropTypes.bool
+	}
+
+	constructor(props) {
+		super(props);
+		const {initialLang, initialOpen} = props;
+		this.state = {
+			currentLangKey: keyLangToLanguageTag(initialLang),
+			currentLangTag: keyLangToLanguageTag(initialLang, false),
+			open: initialOpen
+		};
+	}
+
+	_handleButtonClick = () => {
+		this.setState(
+			prevState => ({
+				open: !prevState.open
+			})
+		);
+	}
+
+	_handleButtonBlur = () => {
+		if (this.state.open) {
+			this.timer = setTimeout(
+				() => {
+					this.setState(
+						() => ({
+							open: false
+						})
+					);
+				},
+				200
+			);
+		}
+	}
+
+	_changeLanguage = langKey => {
+		this.setState(
+			{
+				currentLangKey: keyLangToLanguageTag(langKey),
+				currentLangTag: keyLangToLanguageTag(langKey, false),
+				open: false
+			},
+			() => this.props.onLanguageChange(langKey)
+		);
+	}
+
+	_handleItemFocus = () => {
+		clearTimeout(this.timer);
+	}
+
+	_handleLanguageClick = langKey => () => this._changeLanguage(langKey)
+
+	_handleLanguageKeyboard = langKey => e => {
+		if (e.keyCode === 13) {
+			this._changeLanguage(langKey);
+		}
+	}
+
+	render() {
+		const {
+			currentLangKey,
+			currentLangTag,
+			open
+		} = this.state;
+		const {
+			availableLanguages,
+			defaultLang
+		} = this.props;
+
+		return (<div className={`dropdown postion-relative lfr-icon-menu ${open ? 'open' : ''}`}>
+			<button
+				aria-expanded="false"
+				aria-haspopup="true"
+				className="btn btn-monospaced btn-secondary dropdown-toggle"
+				data-testid="localized-dropdown-button"
+				onBlur={this._handleButtonBlur}
+				onClick={this._handleButtonClick}
+				role="button"
+				title=""
+				type="button"
+			>
+				<span className="inline-item">
+					<ClayIcon iconName={currentLangKey} />
+				</span>
+				<span className="btn-section">{currentLangTag}</span>
+			</button>
+			{
+				open &&
```

Comment:
For rendering conditional jsx blocks, lets try to follow the formatting style to look something like

```jsx
<div>
     {someBoolean && (
           <div>
                 <SomeComponent />
           </div>
     )}
</div>
```